sklearn/multioutput.py

```diff
@@ -812,25 +812,30 @@ def decision_function(self, X):
 
         Returns
         -------
-        Y_decision : array-like of shape (n_samples, n_classes)
+        Y_decision : nd-array-like of shape (n_samples, n_classes, n_outputs)
```

Comment:
This shape is not valid because in multioutput multiclass problems, each multiclass subproblem can have a different value for `n_classes`.

Have a look at the docstring of the `predict_proba` method of another model that supports multioutput multiclass, for instance:

- https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html#sklearn.tree.DecisionTreeClassifier.predict_proba

- https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html#sklearn.neighbors.KNeighborsClassifier.predict_proba\

You may also want to read the tests for the multioutput multiclass case for those estimators.

Also: minor comment: either say ndarray or array-like but not a mix of both. I think for return statements, being precise (`ndarray`) is good.

Reply:
> 

Thanks, @ogrisel 

Now I understood, my changes will work only for homogenous multioutput multiclass. However, after reading the document it is quite possible that these multioutput multiclass could be heterogeneous also. Below I have pasted the code which I think could be suitable for such a case.

from sklearn.multioutput import ClassifierChain
from sklearn.linear_model import LogisticRegression
from sklearn.datasets import make_classification

number_of_classifiers = 3
num_class= [i+2 for i in range (number_of_classifiers)]

Xs, ys = zip(*(make_classification(random_state=i, n_classes=num_class[i], n_informative=3) for i in range(number_of_classifiers)))
X = np.hstack(Xs)
Y = np.transpose(ys)
ClassifierChain(LogisticRegression()).fit(X, Y).decision_function(X)

A possible solution could be a list of (n_samples, n_classes).

Many thanks to you, This is my first contribution and I have good Learning.